src/flow/st-cuts.c

```diff
@@ -1439,6 +1410,26 @@ igraph_error_t igraph_all_st_mincuts(const igraph_t *graph, igraph_real_t *value
     IGRAPH_CHECK(igraph_contract_vertices(&residual, /*mapping=*/ &NtoL, /*vertex_comb=*/ NULL));
     IGRAPH_CHECK(igraph_simplify(&residual, /*multiple=*/ true, /*loops=*/ true, /*edge_comb=*/ NULL));
 
+    /* We relabel the residual graph so that it is in topological sort order. */
+    igraph_vector_int_t order;
+    igraph_t tmpgraph;
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&order, no_of_nodes);
+    IGRAPH_CHECK(igraph_topological_sorting(&residual, &order, IGRAPH_OUT));
+
+    // Relabel mapping
+    for (i = 0; i < igraph_vcount(&residual); i++)
+        VECTOR(NtoL)[i] = VECTOR(order)[VECTOR(NtoL)[i]];
```

Comment:
```suggestion
    for (i = 0; i < igraph_vcount(&residual); i++) {
        VECTOR(NtoL)[i] = VECTOR(order)[VECTOR(NtoL)[i]];
    }
```

Nitpick: We always use braces after `for`, `if`, `while`, etc. It's not my personal preference either, but since we were doing it consistently, let's keep to it.